Detectors/TPC/calibration/src/IDCGroupingParameter.cxx

```diff
@@ -10,7 +10,65 @@
 // or submit itself to any jurisdiction.
 
 #include "TPCCalibration/IDCGroupingParameter.h"
+#include "Framework/Logger.h"
+#include <boost/lexical_cast.hpp>
+#include <boost/tokenizer.hpp>
 
 using namespace o2::tpc;
 O2ParamImpl(o2::tpc::ParameterIDCGroup);
 O2ParamImpl(o2::tpc::ParameterIDCCompression);
+
+void o2::tpc::ParameterIDCGroup::setGroupingParameterFromString(const std::string sgroupPads, const std::string sgroupRows, const std::string sgroupLastRowsThreshold, const std::string sgroupLastPadsThreshold)
+{
+  // convert input string to std::vector<unsigned char>
+  const boost::char_separator<char> sep(","); /// char separator for the tokenizer
+  const boost::tokenizer<boost::char_separator<char>> tgroupPads(sgroupPads, sep);
+  const boost::tokenizer<boost::char_separator<char>> tgroupRows(sgroupRows, sep);
+  const boost::tokenizer<boost::char_separator<char>> tgroupLastRowsThreshold(sgroupLastRowsThreshold, sep);
+  const boost::tokenizer<boost::char_separator<char>> tgroupLastPadsThreshold(sgroupLastPadsThreshold, sep);
+
+  std::vector<unsigned char> vgroupPads;
+  std::vector<unsigned char> vgroupRows;
+  std::vector<unsigned char> vgroupLastRowsThreshold;
+  std::vector<unsigned char> vgroupLastPadsThreshold;
+  vgroupPads.reserve(Mapper::NREGIONS);
+  vgroupRows.reserve(Mapper::NREGIONS);
+  vgroupLastRowsThreshold.reserve(Mapper::NREGIONS);
+  vgroupLastPadsThreshold.reserve(Mapper::NREGIONS);
+
+  std::transform(tgroupPads.begin(), tgroupPads.end(), std::back_inserter(vgroupPads), &boost::lexical_cast<int, std::string>);
+  std::transform(tgroupRows.begin(), tgroupRows.end(), std::back_inserter(vgroupRows), &boost::lexical_cast<int, std::string>);
+  std::transform(tgroupLastRowsThreshold.begin(), tgroupLastRowsThreshold.end(), std::back_inserter(vgroupLastRowsThreshold), &boost::lexical_cast<int, std::string>);
+  std::transform(tgroupLastPadsThreshold.begin(), tgroupLastPadsThreshold.end(), std::back_inserter(vgroupLastPadsThreshold), &boost::lexical_cast<int, std::string>);
```

Comment:
I think you can use the `RangeTokenizer` (https://github.com/AliceO2Group/AliceO2/blob/dev/Algorithm/include/Algorithm/RangeTokenizer.h):
```c++
#include "Algorithm/RangeTokenizer.h"
auto arr = o2::RangeTokenizer::tokenize<unsigned char>("1,2,3,4,5");
arr
(std::vector<unsigned char, std::allocator<unsigned char> > &) { '1', '2', '3', '4', '5' }
```

Reply:
Thanks for the comment. I switched to the RangeTokenizer as suggested